app/models/base_facility.rb

```diff
@@ -139,7 +139,18 @@ def sti_name
     def pull_source_data
       metadata = Facilities::MetadataClient.new.get_metadata(FACILITY_SORT_FIELDS[name].first)
       max_record_count = metadata['maxRecordCount']
-      Facilities::Client.new.get_all_facilities(*FACILITY_SORT_FIELDS[name], max_record_count).map(&method(:new))
+      raw_array = Facilities::Client.new.get_all_facilities(*FACILITY_SORT_FIELDS[name], max_record_count)
```

Comment:
Take a look at the facility_parser middleware and see what you think about pushing this transformation logic in with the other stuff we do to build out this data.  https://github.com/department-of-veterans-affairs/vets-api/blob/cf62a5a9b7b37ad4fdceeda6853a121f0b0d88f5/lib/common/client/middleware/response/facility_parser.rb#L41

Reply:
Had no idea this existed 💡

Makes much more sense to put it there IMO!

Reply:
Thanks, @dhall , it definitely makes sense to put this in the parser.